addons/mail/static/src/models/thread_cache/thread_cache.js

```diff
@@ -282,27 +282,35 @@ function factory(dependencies) {
             const moderated_channel_ids = this.thread.moderation
                 ? [this.thread.id]
                 : undefined;
-            const messages = await this.async(() =>
-                this.env.models['mail.message'].performRpcMessageFetch(
-                    domain,
-                    limit,
-                    moderated_channel_ids,
-                    context,
-                )
-            );
-            this.update({
-                fetchedMessages: [['link', messages]],
-                isLoaded: true,
-                isLoading: false,
-            });
-            if (!extraDomain && messages.length < limit) {
-                this.update({ isAllHistoryLoaded: true });
+            try {
+                const messages = await this.async(() =>
+                    this.env.models['mail.message'].performRpcMessageFetch(
+                        domain,
+                        limit,
+                        moderated_channel_ids,
+                        context,
+                    )
+                );
+                this.update({
+                    fetchedMessages: [['link', messages]],
+                    isLoaded: true,
+                    isLoading: false,
+                });
+                if (!extraDomain && messages.length < limit) {
+                    this.update({ isAllHistoryLoaded: true });
+                }
+                this.env.messagingBus.trigger('o-thread-cache-loaded-messages', {
+                    fetchedMessages: messages,
+                    threadCache: this,
+                });
+                return messages;
+            } catch(e) {
+                this.update({
+                    isLoaded: true,
```

Comment:
Setting `isLoaded` `true` when it's not loaded due to a crash is not correct actually...
I guess we need a new field for the error state? And specific display in view? And maybe auto-retry after some time?

I assume by doing current fix but keeping `isLoaded` to `false` it is retrying immediately forever?

Reply:
exactly, infinite loop

Reply:
So do you see a solution that would work without having to set `isLoaded: true` when it failed to load? One of the ways I mentioned above?

Reply:
I think it's kind of overkill to handle a retry after some time, but we could maybe add a field to handle errors and show that there has been a problem while loading messages. By the way, this new field would make any loading logic more complex as it will be needed to check if thread is loaded or is in error.

Reply:
It's not really overkill though, if there is a server timeout while the fetch was in progress, it will currently not be possible ever to fetch the messages (except full refresh page), and worse the display will be as if the thread is empty.

Reply:
Conceptually I think it's even "better" to be stuck in "is loading" as before this PR, as it would be to be stuck in "loaded, but not really".